File: OOP/Homeworks/Homework-09-OOP-in-Java/01.Geometry/src/geometry/geometry2d/Circle.java
package geometry.geometry2d;

import geometry.Vertex;

public class Circle extends PlaneShape {
    private double radius;

    public Circle (Vertex2D center, double radius) {
        super(center);
        this.setRadius(radius);
    }

    public double getRadius() {
        return this.radius;
    }

    public void setRadius(double radius) {
        if (radius <= 0) {
            throw new IllegalArgumentException(
                    "The radius of a circle should be greater than zero.");
        }

        this.radius = radius;
    }

    public Vertex getCenter() {
        return this.vertices.get(0);
    }

    @Override
    public double getArea() {
        double area = Math.PI * this.getRadius() * this.getRadius();

        return area;
    }

    @Override
    public double getPerimeter() {
        double perimeter = 2 * Math.PI * this.getRadius();

        return perimeter;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Circle" + System.lineSeparator());
        sb.append(String.format("Center: %s%n", this.getCenter()));
        sb.append(String.format("Radius: %.2f", this.getRadius()));
        sb.append(super.toString());

        return sb.toString();
    }
}

File: OOP/Homeworks/Homework-04-Encapsulation-Polymorphism/01.Shapes/Shapes/Circle.cs
namespace _01.Shapes.Shapes
{
    using System;
    using Contracts;

    class Circle : IShape
    {
        private double radius;

        public Circle(double radius)
        {
            this.Radius = radius;
        }

        public double Radius
        {
            get
            {
                return this.radius;
            }

            set
            {
                if (value <= 0)
                {
                    throw new ArgumentOutOfRangeException("radius", "The radius should be greater than zero.");
                }

                this.radius = value;
            }
        }

        public double CalculateArea()
        {
            double area = Math.PI * this.Radius * this.Radius;
            return area;
        }

        public double CalculatePerimeter()
        {
            double perimeter = 2 * Math.PI * this.Radius;
            return perimeter;
        }
    }
}
